### DISCUSSION TITLE
pwm on mimxrt1064_evk

### DISCUSSION THREAD

**Discussion by Chethan-TS** [zephyrproject-rtos/zephyr#80070] [General] [answered]
hi,

i am trying a servo example with mimxrt1064_evk board where as per the official documentation on zephyr RTOS supports PWM on the mentioned board, 

![image](https://github.com/user-attachments/assets/2ff3ed0c-f0b1-4c01-806f-376283359298)

i have written a separate overlay for the same , i have uploaded all the files for verification 

[servo_motor_test.zip](https://github.com/user-attachments/files/17432174/servo_motor_test.zip)

but when i build the project i keep getting the following error,

west build -p -b mimxrt1064_evk
-- west build: making build dir /home/chethan/GCB_rnd/BLDC/1064/servo_motor_test/build pristine
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: /home/chethan/GCB_rnd/BLDC/1064/servo_motor_test
-- CMake version: 3.29.6
-- Found Python3: /usr/bin/python3 (found suitable version "3.10.12", minimum required is "3.8") found components: Interpreter
-- Cache files will be written to: /home/chethan/.cache/zephyr
-- Zephyr version: 3.7.0-rc2 (/home/chethan/zephyrproject/zephyr)
-- Found west (found suitable version "1.2.0", minimum required is "0.14.0")
-- Board: mimxrt1064_evk, qualifiers: mimxrt1064
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 0.16.5 (/home/chethan/zephyr-sdk-0.16.5-1)
-- Found toolchain: zephyr 0.16.5 (/home/chethan/zephyr-sdk-0.16.5-1)
-- Found Dtc: /home/chethan/zephyr-sdk-0.16.5-1/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6")
-- Found BOARD.dts: /home/chethan/zephyrproject/zephyr/boards/nxp/mimxrt1064_evk/mimxrt1064_evk.dts
-- Found devicetree overlay: /home/chethan/GCB_rnd/BLDC/1064/servo_motor_test/app.overlay
-- Generated zephyr.dts: /home/chethan/GCB_rnd/BLDC/1064/servo_motor_test/build/zephyr/zephyr.dts
-- Generated devicetree_generated.h: /home/chethan/GCB_rnd/BLDC/1064/servo_motor_test/build/zephyr/include/generated/zephyr/devicetree_generated.h
-- Including generated dts.cmake file: /home/chethan/GCB_rnd/BLDC/1064/servo_motor_test/build/zephyr/dts.cmake
Parsing /home/chethan/zephyrproject/zephyr/Kconfig
Loaded configuration '/home/chethan/zephyrproject/zephyr/boards/nxp/mimxrt1064_evk/mimxrt1064_evk_defconfig'
Merged configuration '/home/chethan/GCB_rnd/BLDC/1064/servo_motor_test/prj.conf'
Configuration saved to '/home/chethan/GCB_rnd/BLDC/1064/servo_motor_test/build/zephyr/.config'
Kconfig header saved to '/home/chethan/GCB_rnd/BLDC/1064/servo_motor_test/build/zephyr/include/generated/zephyr/autoconf.h'
-- Found GnuLd: /home/chethan/zephyr-sdk-0.16.5-1/arm-zephyr-eabi/arm-zephyr-eabi/bin/ld.bfd (found version "2.38")
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /home/chethan/zephyr-sdk-0.16.5-1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
Load components for MIMXRT1064:
driver_common component is included.
driver_reset component is included.
device_CMSIS component is included.
CMSIS_Include_core_cm component is included.
device_system component is included.
driver_lpuart component is included.
driver_igpio component is included.
driver_pwm component is included.
driver_cache_armv7_m7 component is included.
-- Using ccache: /usr/bin/ccache
-- Configuring done (7.1s)
-- Generating done (0.1s)
-- Build files have been written to: /home/chethan/GCB_rnd/BLDC/1064/servo_motor_test/build
-- west build: building application
[1/135] Preparing syscall dependency handling

[3/135] Generating include/generated/zephyr/version.h
-- Zephyr version: 3.7.0-rc2 (/home/chethan/zephyrproject/zephyr), build: v3.7.0-rc2-326-g6fc525cc20d4
[18/135] Building C object CMakeFiles/app.dir/src/main.c.obj
FAILED: CMakeFiles/app.dir/src/main.c.obj 
ccache /home/chethan/zephyr-sdk-0.16.5-1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DBOARD_FLASH_SIZE="CONFIG_FLASH_SIZE*1024" -DCPU_MIMXRT1064DVL6A -DKERNEL -DK_HEAP_MEM_POOL_SIZE=0 -DNDEBUG -DPICOLIBC_LONG_LONG_PRINTF_SCANF -DXIP_BOOT_HEADER_DCD_ENABLE=1 -DXIP_BOOT_HEADER_ENABLE=1 -DXIP_EXTERNAL_FLASH -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR__=1 -I/home/chethan/GCB_rnd/BLDC/1064/servo_motor_test/build/zephyr/include/generated/zephyr -I/home/chethan/zephyrproject/zephyr/include -I/home/chethan/GCB_rnd/BLDC/1064/servo_motor_test/build/zephyr/include/generated -I/home/chethan/zephyrproject/zephyr/soc/nxp/imxrt -I/home/chethan/zephyrproject/zephyr/soc/nxp/imxrt/imxrt10xx/. -I/home/chethan/zephyrproject/zephyr/soc/nxp/imxrt/. -I/home/chethan/zephyrproject/zephyr/soc/nxp/imxrt/imxrt10xx -I/home/chethan/zephyrproject/modules/hal/cmsis/CMSIS/Core/Include -I/home/chethan/zephyrproject/zephyr/modules/cmsis/. -I/home/chethan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT1064 -I/home/chethan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT1064/drivers -I/home/chethan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT1064/xip -I/home/chethan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/common -I/home/chethan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/lpuart -I/home/chethan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/igpio -I/home/chethan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/pwm -I/home/chethan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/cache/armv7-m7 -I/home/chethan/zephyrproject/zephyr/modules/hal_nxp/. -isystem /home/chethan/zephyrproject/zephyr/lib/libc/common/include -fno-strict-aliasing -Os -imacros /home/chethan/GCB_rnd/BLDC/1064/servo_motor_test/build/zephyr/include/generated/zephyr/autoconf.h -fno-printf-return-value -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m7 -mthumb -mabi=aapcs -mfp16-format=ieee -mtp=soft --sysroot=/home/chethan/zephyr-sdk-0.16.5-1/arm-zephyr-eabi/arm-zephyr-eabi -imacros /home/chethan/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/chethan/GCB_rnd/BLDC/1064/servo_motor_test=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/chethan/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/chethan/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections --specs=picolibc.specs -std=c99 -MD -MT CMakeFiles/app.dir/src/main.c.obj -MF CMakeFiles/app.dir/src/main.c.obj.d -o CMakeFiles/app.dir/src/main.c.obj -c /home/chethan/GCB_rnd/BLDC/1064/servo_motor_test/src/main.c
/home/chethan/GCB_rnd/BLDC/1064/servo_motor_test/src/main.c:67:2: error: unterminated argument list invoking macro "PWM_DT_SPEC_GET"
   67 | }
      |  ^
/home/chethan/GCB_rnd/BLDC/1064/servo_motor_test/src/main.c:16:41: error: 'PWM_DT_SPEC_GET' undeclared here (not in a function)
   16 | static const struct pwm_dt_spec servo = PWM_DT_SPEC_GET(DT_NODELABEL(servo);
      |                                         ^~~~~~~~~~~~~~~
/home/chethan/GCB_rnd/BLDC/1064/servo_motor_test/src/main.c:16:21: error: expected ',' or ';' at end of input
   16 | static const struct pwm_dt_spec servo = PWM_DT_SPEC_GET(DT_NODELABEL(servo);
      |                     ^~~~~~~~~~~
/home/chethan/GCB_rnd/BLDC/1064/servo_motor_test/src/main.c:16:33: warning: 'servo' defined but not used [-Wunused-const-variable=]
   16 | static const struct pwm_dt_spec servo = PWM_DT_SPEC_GET(DT_NODELABEL(servo);
      |                                 ^~~~~
[31/135] Building C object zephyr/arch/arch/arm/core/CMakeFiles/arch__arm__core.dir/fatal.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /home/chethan/GCB_rnd/BLDC/1064/servo_motor_test/build

--------------------------------------------------------------------------------------------------------------------------------------------
i don't understand the reason behind as far as i am aware i get the above error when i am building for unsupported board, which i am aware for the documentation for blinky example.

![image](https://github.com/user-attachments/assets/cc50124e-86e0-47f2-981d-69acf12d3435)
 
if i can get assistance for the same , it would be helpful.

PS:  *version :  main

thanks in advance,
Chethan T S

**Comment by Kaleb-Belete** [comment, marked as answer]
Hi Chethan,
The error I noticed in your code is due to not closing the parenthesis in "PWM_DT_SPEC_GET" in the main.c file (line 16).

Regards, 
Kaleb Belete
